napari/layers/_shapes_layer/model.py

```diff
@@ -1148,7 +1233,142 @@ def to_xml_list(self, shape_type=None):
             List of xml elements defining each shape according to the
             svg specification
         """
-        return self.data.to_xml_list(shape_type=shape_type)
+        return self._data_view.to_xml_list(shape_type=shape_type)
```

Comment:
After reading a bunch of this code, I am slightly wondering whether `_data_view` should (a) be made public, and (b) be renamed `slice_data`. The latter is more descriptive, whereas "data view" could be anything really. And I also am weary of using *lots* of private attributes and APIs internally, which makes it hard for e.g. plugins to access the same kind of functionality that we do.

Reply:
done